preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py

```diff
@@ -198,6 +201,163 @@ def check_date(
                 ],
             )
 
+    @staticmethod
+    def parse_date_into_range(
+        input_data: InputMetadata,
+        output_field: str,
+        args: FunctionArgs = None,  # args is essential - even if Pylance says it's not used
+    ) -> ProcessingResult:
+        """Parse date string into a range, return based on FunctionArgs
+        Three output options
+        fieldType: "dateRangeString" | "dateRangeLower" | "dateRangeUpper"
+        """
+        logger.debug(f"input_data: {input_data}")
+        date_str = input_data["date"]
+
+        release_date_str = input_data.get("releaseDate", "") or ""
+        try:
+            release_date = dateutil.parse(release_date_str).astimezone(pytz.utc)
+        except Exception:
+            release_date = None
+
+        max_upper_limit = min(filter(None, [datetime.now(tz=pytz.utc), release_date]))
```

Comment:
Thanks, yes that's simpler now. There was originally a third field that could be none, but I removed that. It's true it could be simplified. In general, on the Python side between Anya and I we have laxer standards regarding review as we are fewer people working on that part (just her and me essentially).